java/trace/src/main/scala/org/typelevel/otel4s/java/trace/WrappedSpanContext.scala

```diff
@@ -24,70 +24,37 @@ import org.typelevel.otel4s.trace.TraceFlags
 import org.typelevel.otel4s.trace.TraceState
 import scodec.bits.ByteVector
 
-private[java] final case class WrappedSpanContext(
-    jSpanContext: JSpanContext
-) extends SpanContext {
+private[otel4s] object WrappedSpanContext {
 
-  lazy val traceId: ByteVector =
-    ByteVector(jSpanContext.getTraceIdBytes)
-
-  def traceIdHex: String =
-    jSpanContext.getTraceId
-
-  lazy val spanId: ByteVector =
-    ByteVector(jSpanContext.getSpanIdBytes)
-
-  def spanIdHex: String =
-    jSpanContext.getSpanId
-
-  lazy val traceFlags: TraceFlags =
-    TraceFlags.fromByte(jSpanContext.getTraceFlags.asByte)
-
-  lazy val traceState: TraceState = {
+  def wrap(context: JSpanContext): SpanContext = {
     val entries = Vector.newBuilder[(String, String)]
-    jSpanContext.getTraceState.forEach((k, v) => entries.addOne(k -> v))
-    TraceState.fromVectorUnsafe(entries.result())
+    context.getTraceState.forEach((k, v) => entries.addOne(k -> v))
+    val traceState = TraceState.fromVectorUnsafe(entries.result())
+
+    SpanContext(
```

Comment:
This is re-validating it right? Probably not a big deal but I suppose we could use some sort of unsafe constructor to skip validation in this specific case.

Reply:
Yes, we can use `SpanContext.createInternal` instead.